app/src/test/java/com/concordium/wallet/data/util/ParamsDeserializerShould.kt

```diff
@@ -0,0 +1,106 @@
+
+        val paramsWithSchemaAsObject = "{\"schema\":{" +
+                "\"type\":\"module\"," +
+                "\"value\":\"//8CAQAAAAkAAABjaXMyX3dDQ0QBABQAAgAAAAMAAAB1cmwWAgQAAABoYXNoFQIAAAAEAAAATm9uZQIEAAAAU29tZQEBAAAAEyAAAAACAQAAAAQAAAB3cmFwBBQAAgAAAAIAAAB0bxUCAAAABwAAAEFjY291bnQBAQAAAAsIAAAAQ29udHJhY3QBAgAAAAwWAQQAAABkYXRhHQEVBAAAAA4AAABJbnZhbGlkVG9rZW5JZAIRAAAASW5zdWZmaWNpZW50RnVuZHMCDAAAAFVuYXV0aG9yaXplZAIGAAAAQ3VzdG9tAQEAAAAVCQAAAAsAAABQYXJzZVBhcmFtcwIHAAAATG9nRnVsbAIMAAAATG9nTWFsZm9ybWVkAg4AAABDb250cmFjdFBhdXNlZAITAAAASW52b2tlQ29udHJhY3RFcnJvcgITAAAASW52b2tlVHJhbnNmZXJFcnJvcgIaAAAARmFpbGVkVXBncmFkZU1pc3NpbmdNb2R1bGUCHAAAAEZhaWxlZFVwZ3JhZGVNaXNzaW5nQ29udHJhY3QCJQAAAEZhaWxlZFVwZ3JhZGVVbnN1cHBvcnRlZE1vZHVsZVZlcnNpb24C\"}}"
+
+        val paramsFromStringSchema = getParameters(paramsWithSchemaAsString)
+        val paramsFromObjectSchema = getParameters(paramsWithSchemaAsObject)
+
+        Assert.assertEquals(
+            paramsFromStringSchema?.schema?.value,
+            paramsFromObjectSchema?.schema?.value
+        )
+    }
+
+    /**
+     * If the parameters have invalid JSON, it should throw [JsonSyntaxException]
+     */
+    @Test
+    fun throwErrorForInvalidJson() {
+        val invalidJson = "not a json"
+        Assert.assertThrows(JsonSyntaxException::class.java) {
+            getParameters(invalidJson)
+        }
+    }
+
+    /**
+     * The Schema should be null if it is not present in the request
+     */
+    @Test
+    fun setSchemaAsNullInResultIfNoSchemaPresentInInput() {
+        val invalidJson = "{\"type\":\"Update\"," +
+                "\"sender\":\"3KW3RkupTdDBomBqHztpMtSt4ZNUv2RdTbcSBgq7onSxg6hXRJ\"}"
+        val params = getParameters(invalidJson)
+
+        Assert.assertNull(params!!.schema)
+    }
+
+    /**
+     * If the input is null, the response should also be null
+     */
+    @Test
+    fun returnNullIfJsonIsNull() {
+        val nullInput: String? = null
+        Assert.assertNull(getParameters(nullInput))
+    }
+
+    /**
+     * If the input is empty, the response should be null
+     */
+    @Test
+    fun returnNullIfJsonIsEmpty() {
+        val emptyInput = ""
+        Assert.assertNull(getParameters(emptyInput))
+    }
+
+    /**
+     * If the schema object in the request is flawed then the schema in the response should be null
+     */
+    @Test
+    fun setSchemaAsNullInResultIfSchemaObjectIsFlowed() {
```

Comment:
Flawed instead of Flowed

Reply:
@ali-concordium There is a nice "suggest change" feature as described [here](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/reviewing-changes-in-pull-requests/commenting-on-a-pull-request#adding-line-comments-to-a-pull-request) (item 6) that allows the author to apply the suggestion directly through the UI. And even nicer is its shortcut `ctrl-g` :)